cas-server-support-oauth/src/main/java/org/jasig/cas/support/oauth/ticket/accesstoken/DefaultAccessTokenFactory.java

```diff
@@ -0,0 +1,70 @@
+package org.jasig.cas.support.oauth.ticket.accesstoken;
+
+import org.jasig.cas.authentication.Authentication;
+import org.jasig.cas.authentication.principal.Service;
+import org.jasig.cas.ticket.ExpirationPolicy;
+import org.jasig.cas.ticket.Ticket;
+import org.jasig.cas.ticket.TicketFactory;
+import org.jasig.cas.ticket.UniqueTicketIdGenerator;
+import org.jasig.cas.util.DefaultUniqueTicketIdGenerator;
+import org.pac4j.core.util.CommonHelper;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.stereotype.Component;
+
+import javax.validation.constraints.NotNull;
+
+/**
+ * Default OAuth access token factory.
+ *
+ * @author Jerome Leleu
+ * @since 4.3.0
+ */
+@Component("defaultAccessTokenFactory")
+public class DefaultAccessTokenFactory implements AccessTokenFactory {
+
+    protected final Logger logger = LoggerFactory.getLogger(this.getClass());
+
+    /** Default instance for the ticket id generator. */
+    @NotNull
+    @Autowired(required = false)
+    @Qualifier("accessTokenIdGenerator")
+    protected UniqueTicketIdGenerator accessTokenIdGenerator = new DefaultUniqueTicketIdGenerator();
+
+
+    /** ExpirationPolicy for access tokens. */
+    @Autowired(required = false)
```

Comment:
Yes, you are completely right. This "trick" was useful to make tests pass when other modules depend on the `cas-server-support-oauth` one. They now depend on the `cas-server-support-oauth-core` module which solves the problem.